pypicloud/access/aws_secrets_manager.py

```diff
@@ -0,0 +1,181 @@
+                raise Exception("The request was invalid due to:", e)
+            elif e.response['Error']['Code'] == 'InvalidParameterException':
+                raise Exception("The request had invalid params:", e)
+
+    def verify_user(self, username, password):
+        credentials = self._fetch_credentials()
+        try:
+            return credentials.get('users', {})[username] == password
+        except KeyError:
+            return False
+
+    def _get_password_hash(self, username):
+        # We don't have to do anything here because we overrode 'verify_user'
+        pass
+
+    def groups(self, username=None):
+        credentials = self._fetch_credentials()
+        if not username:
+            return list(credentials.get('groups', {}).keys())
+        groups = []
+        for group_name, users in credentials.get('groups', {}).items():
+            if username in users:
+                groups.append(group_name)
+        return groups
+
+    def group_members(self, group):
+        credentials = self._fetch_credentials()
+        return list(credentials.get('groups', {}).get(group, []))
+
+    def is_admin(self, username):
+        credentials = self._fetch_credentials()
+        admins = credentials.get('groups', {}).get('admins', [])
+        return username in admins
+
+    def group_permissions(self, package):
+        credentials = self._fetch_credentials()
+        packages = credentials.get('packages', {})
+        package = packages.get('package', {})
+        return package.get('groups', {})
+
+    def user_permissions(self, package):
+        credentials = self._fetch_credentials()
+        packages = credentials.get('packages', {})
+        package = packages.get('package', {})
+        return package.get('users', {})
+
+    def user_package_permissions(self, username):
+        credentials = self._fetch_credentials()
+        packages = []
+        for package in credentials.get('packages', {}):
+            users = package.get('users', {})
+            if username in users:
+                packages.append(username)
+        return packages
+
+    def group_package_permissions(self, group):
+        credentials = self._fetch_credentials()
+        packages = []
+        for package in credentials.get('packages', {}):
+            groups = package.get('groups', {})
+            if group in groups:
+                packages.append(group)
+        return packages
+
+    def user_data(self, username=None):
+        credentials = self._fetch_credentials()
+        admins = credentials.get('groups', {}).get('admins', [])
+        users = []
+        for user in credentials.get('users', {}):
+            if username and user != username:
+                continue
+            users.append({
+                'username': user,
+                'admin': user in admins
+            })
+            if username and user == username:
+                break
```

Comment:
You'll need to return a dict here that includes the groups that the user is in. The API could be better, I know. You can find the documentation here https://github.com/stevearc/pypicloud/blob/e9b1d3d284a65f9ad8767b64044c2b1bbeecc315/pypicloud/access/base.py#L412-L418